lib/phlex/rails/layout.rb

```diff
@@ -12,5 +12,37 @@ module Layout
 		include Helpers::JavaScriptIncludeTag
 		include Helpers::JavaScriptImportMapTags
 		include Helpers::JavaScriptImportModuleTag
+
+		module ClassMethods
+			def render(view, _locals, &block)
+				component = allocate
+
+				component.call(view_context: view) do |yielded|
+					case yielded
+					when Symbol
+						output = view.view_flow.get(yielded)
+					else
+						output = yield(yielded)
+					end
+
+					case output
+					when ActiveSupport::SafeBuffer
+						component.unsafe_raw output
+					end
+				end
+			end
+
+			def identifier
+				name
+			end
+
+			def virtual_path
+				""
+			end
+		end
+
+		def self.included(klass)
+			klass.extend(ClassMethods)
```

Comment:
I wonder if this `ClassMethods` module should be `Phlex::Rails::Layout` and it should be _extended_ rather than _included_.

This _included_ callback won't work if the module is included into another module before being included into the class.

Reply:
The other option is to make it a class, e.g.

```ruby
class ApplicationLayout < Phlex::Rails::Layout
  ...
end
```

If necessary, `ApplicationLayout` and `ApplicationView` could share stuff though a module. e.g.

```ruby
ApplicationView.include(SharedHelpers)
ApplicationLayout.include(SharedHelpers)
```

Reply:
> This included callback won't work if the module is included into another module before being included into the class.

Maybe it would make sense to check if the current class the modules gets included in has a `def template` method, and if it doesn't it could show a warning that you can't you directly need to include the module.